tools/api.py

```diff
@@ -308,18 +311,36 @@ def virtual_param(self, prefix: Optional[str] = None) -> str:
         return self.substitute('${type} ${name}', prefix=prefix)
 
     def virtual_output_args(self, prefix: Optional[str] = None) -> List[str]:
-        return [
-            '&{prefix}{n}'.format(prefix=prefix or '', n=n)
-            for t, n in self.cparams
-        ]
+        container_type = self.type.remove_generic().basic().str()
+        decl_list: List[str] = []
+        for t, n, in self.cparams:
+            if not decl_list and (container_type == "std::vector"
+                                  or container_type == "vector"):
+                decl_list.append('{prefix}{n}.data()'.format(prefix=prefix
+                                                             or '',
+                                                             n=n))
+            else:
+                decl_list.append('&{prefix}{n}'.format(prefix=prefix or '',
+                                                       n=n))
+        return decl_list
 
     def virtual_output_declarations(self,
                                     prefix: Optional[str] = None) -> List[str]:
-        return [
-            'std::remove_pointer_t<{type}> {prefix}{n};'.format(
-                type=Type(t).str(), prefix=prefix or '', n=n)
-            for t, n in self.cparams
-        ]
+        container_type = self.type.remove_generic().basic().str()
+        decl_list: List[str] = []
+        for t, n, in self.cparams:
+            if not decl_list and (container_type == "std::vector"
+                                  or container_type == "vector"):
+                inner_t = self.type.inner_type()
+                if inner_t:
+                    decl_list.append(
+                        'std::vector<{inner_t}> {prefix}{n}(1024);'.format(
```

Comment:
This `1024` is a constant we could declare this as array instead: `std::array<{inner_t}, 1024> {prefix}{n};`

Reply:
Done.